From: Ben Hutchings <ben@decadent.org.uk>
Date: Thu, 02 Jan 2020 17:44:46 +0000
Subject: ipvs: Avoid ABI change in 3.16.80
Forwarded: not-needed

Commit 74b5e1077f6d "ipvs: move old_secure_tcp into struct netns_ipvs"
added a new member to struct netns_ipvs.  This structure is visible to
module.

The structure is only allocated, and the new member used, in the ipvs
module.  So it is safe to add the new member at the end.  Do that and
hide it from genksyms.

---
--- a/include/net/ip_vs.h
+++ b/include/net/ip_vs.h
@@ -919,7 +919,6 @@ struct netns_ipvs {
 	struct delayed_work	defense_work;   /* Work handler */
 	int			drop_rate;
 	int			drop_counter;
-	int			old_secure_tcp;
 	atomic_t		dropentry;
 	/* locks in ctl.c */
 	spinlock_t		dropentry_lock;  /* drop entry handling */
@@ -987,6 +986,9 @@ struct netns_ipvs {
 	char			backup_mcast_ifn[IP_VS_IFNAME_MAXLEN];
 	/* net name space ptr */
 	struct net		*net;            /* Needed by timer routines */
+#ifndef __GENKSYMS__
+	int			old_secure_tcp;
+#endif
 };
 
 #define DEFAULT_SYNC_THRESHOLD	3
